From b3e069d32994be787abcaacf5e796953b64e10cb Mon Sep 17 00:00:00 2001
From: Kihwal Lee <kihwal@apache.org>
Date: Mon, 9 May 2016 15:50:09 -0500
Subject: [PATCH 1524/2748] HDFS-10372. Fix for failing
 TestFsDatasetImpl#testCleanShutdownOfVolume.
 Contributed by Rushabh Shah.

(cherry picked from commit b9e5a32fa14b727b44118ec7f43fb95de05a7c2c)

Change-Id: If35c1179f9a3a2dfe741b32ce8de4216f53654be
---
 .../datanode/fsdataset/impl/TestFsDatasetImpl.java |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/TestFsDatasetImpl.java b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/TestFsDatasetImpl.java
index 47faffb..40ee8f7 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/TestFsDatasetImpl.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/TestFsDatasetImpl.java
@@ -584,7 +584,7 @@ public void testCleanShutdownOfVolume() throws Exception {
         Assert.fail("This is not a valid code path. "
             + "out.close should have thrown an exception.");
       } catch (IOException ioe) {
-        GenericTestUtils.assertExceptionContains(info.toString(), ioe);
+        GenericTestUtils.assertExceptionContains(info.getXferAddr(), ioe);
       }
       finalizedDir.setWritable(true);
       finalizedDir.setExecutable(true);
-- 
1.7.9.5

